### DISCUSSION TITLE
How to Integrate Great Expectations with MongoDB?

### DISCUSSION THREAD

**Discussion by h2cone** [great-expectations/great_expectations#10008] [Support] [answered]
I am currently working on a project that uses MongoDB as our primary database, and I'm very interested in leveraging Great Expectations to validate and ensure the quality of our data. However, I am unsure of the best approach to integrate Great Expectations with MongoDB.

Could someone please provide guidance or examples on how to set up this integration?

**Comment by Kilo59** [comment, marked as answer]
@h2cone This isn't supported natively.

You would have to [load your data/documents](https://pymongo.readthedocs.io/en/stable/tutorial.html#querying-for-more-than-one-document) into a pandas or spark dataframe (or a [file format that they support](https://pandas.pydata.org/docs/user_guide/io.html#io-tools-text-csv-hdf5)) and then you could use a Spark/Pandas datasource.

https://docs.greatexpectations.io/docs/0.18/oss/guides/connecting_to_your_data/fluent/in_memory/connect_in_memory_data
https://docs.greatexpectations.io/docs/0.18/oss/guides/connecting_to_your_data/fluent/filesystem/connect_filesystem_source_data
